packages/replay/src/util/handleRecordingEmit.ts

```diff
@@ -84,3 +93,38 @@ export function getHandleRecordingEmit(replay: ReplayContainer): RecordingEmitCa
     });
   };
 }
+
+/**
+ * Add an event to the event buffer.
+ * `isCheckout` is true if this is either the very first event, or an event triggered by `checkoutEveryNms`.
+ */
+export async function addSettingsEvent(replay: ReplayContainer, isCheckout?: boolean): Promise<void | null> {
+  // Only need to add this event when sending the first segment
+  if (!isCheckout || !replay.session || replay.session.segmentId !== 0) {
+    return null;
+  }
+
+  const options = replay.getOptions();
+  const event = {
+    type: EventType.Custom,
+    timestamp: new Date().getTime(),
+    data: {
+      tag: 'options',
+      payload: {
+        sessionSampleRate: options.sessionSampleRate,
+        errorSampleRate: options.errorSampleRate,
+        useCompressionOption: options.useCompression,
+        blockAllMedia: options.blockAllMedia,
+        maskAllText: options.maskAllText,
+        maskAllInputs: options.maskAllInputs,
+        useCompression: replay.eventBuffer && replay.eventBuffer.type === 'worker',
```

Comment:
```suggestion
        useCompression: replay.eventBuffer ? replay.eventBuffer.type === 'worker' : false,
```

To ensure this is always a boolean?